src/decoder/mod.rs

```diff
@@ -1390,43 +1478,44 @@ impl<R: Read + Seek> Decoder<R> {
     }
 
     /// Read a single strip from the image and return it as a Vector
-    pub fn read_strip(&mut self) -> TiffResult<DecodingResult> {
-        self.check_chunk_type(ChunkType::Strip)?;
-        let index = self.current_chunk;
+    pub fn read_strip(&mut self) -> TiffResult<(DecodingResult, ChunkInfo)> {
```

Comment:
The comment regarding `ChunkInfo` was only referring to new interfaces, this would be a breaking change. Please revert the interface change here and in `read_tile`, and only in those two instances. 

This is a loss of information because it drops information in the case of the last strip (which need not contain the full `rows_per_strip` rows but might be less) and border tiles. This is slightly unfortunate, so adding a comment that this interface could be changed in a future release would be nice.

Reply:
Yeah I should have realised that is what you meant - I was surprised at the suggested change.